modules/auxiliary/scanner/http/git_scanner.rb

```diff
@@ -0,0 +1,131 @@
+  include Msf::Auxiliary::WmapScanServer
+  # Scanner mixin should be near last
+  include Msf::Auxiliary::Scanner
+  include Msf::Auxiliary::Report
+
+  def initialize
+    super(
+      'Name'        => 'HTTP Git Scanner',
+      'Description' => 'Detect git directories and files and analize its content.',
+      'Author'       => ['t0nyhj'],
+      'License'     => MSF_LICENSE
+    )
+
+    register_options(
+      [
+        OptString.new('PATH', [ true,  "The test path to .git directory", '/'])#,
+        #OptBool.new('GET_SOURCE', [ false, "Attempt to obtain file source code", true ]),
+        #OptBool.new('SHOW_SOURCE', [ false, "Show source code", true ])
+
+      ], self.class)
+
+    register_advanced_options(
+      [
+        OptInt.new('ErrorCode', [ true, "Error code for non existent directory", 404]),
+        OptPath.new('HTTP404Sigs',   [ false, "Path of 404 signatures to use",
+            File.join(Msf::Config.data_directory, "wmap", "wmap_404s.txt")
+          ]
+        ),
+        OptBool.new('NoDetailMessages', [ false, "Do not display detailed test messages", true ])
+
+      ], self.class)
+  end
+
+  def run_host(target_host)
+    conn = true
+    ecode = nil
+    emesg = nil
+
+    word1 = 'core'
+    word2 = 'remote'
+    word3 = 'branch'
+
+    tpath = normalize_uri(datastore['PATH'])
+    if tpath[-1,1] != '/'
+      tpath += '/'
+    end
+
+    ecode = datastore['ErrorCode'].to_i
+    vhost = datastore['VHOST'] || wmap_target_host
+
+    #
+    # Detect error code
+    #
+    begin
+      randdir = Rex::Text.rand_text_alpha(5).chomp + '/'
+      res = send_request_cgi({
+        'uri'  		=>  tpath+randdir,
+        'method'   	=> 'GET',
+        'ctype'		=> 'text/html'
+      }, 20)
+
+      return if not res
+
+      tcode = res.code.to_i
+
+
+      # Look for a string we can signature on as well
+      if(tcode >= 200 and tcode <= 299)
+
+        File.open(datastore['HTTP404Sigs'], 'rb').each do |str|
+          if(res.body.index(str))
+            emesg = str
+            break
+          end
+        end
+
+        if(not emesg)
+          #print_status("Using first 256 bytes of the response as 404 string")
+          emesg = res.body[0,256]
+        else
+          #print_status("Using custom 404 string of '#{emesg}'")
+        end
+      else
+        ecode = tcode
+        #print_status("Using code '#{ecode}' as not found.")
+      end
+
+    rescue ::Rex::ConnectionRefused, ::Rex::HostUnreachable, ::Rex::ConnectionTimeout
+      conn = false
+    rescue ::Timeout::Error, ::Errno::EPIPE
+    end
+
+    return if not conn
+
+    dm = datastore['NoDetailMessages']
+
+    begin
+      turl = tpath+'.git/config'#'.svn/entries'
+
+      res = send_request_cgi({
+        'uri'          => turl,
+        'method'       => 'GET',
+        'version' => '1.0',
+      }, 10)
+
+      if(not res or ((res.code.to_i == ecode) or (emesg and res.body.index(emesg))))
+        if dm == false
```

Comment:
When the VERBOSE to be true, it will lose some output. Like below:

msf > use auxiliary/scanner/http/git_scanner
msf auxiliary(git_scanner) > set rhosts 180.150.151.130/24
rhosts => 180.150.151.130/24
msf auxiliary(git_scanner) > set threads 300
threads => 300
msf auxiliary(git_scanner) > run

[+] [180.150.151.158:80] Git Config file found.
[+] [180.150.151.131:80] Git Config file found.
[+] [180.150.151.130:80] Git Config file found.
[+] [180.150.151.128:80] Git Config file found.
[*] Scanned  84 of 256 hosts (32% complete)
[*] Scanned  97 of 256 hosts (37% complete)
[*] Scanned  98 of 256 hosts (38% complete)
[*] Scanned 110 of 256 hosts (42% complete)
[*] Scanned 173 of 256 hosts (67% complete)
[*] Scanned 206 of 256 hosts (80% complete)
[*] Scanned 208 of 256 hosts (81% complete)
[*] Scanned 229 of 256 hosts (89% complete)
[*] Scanned 240 of 256 hosts (93% complete)
[*] Scanned 256 of 256 hosts (100% complete)
[*] Auxiliary module execution completed
msf auxiliary(git_scanner) > set VERBOSE true
VERBOSE => true
msf auxiliary(git_scanner) > run

[*] Scanned  82 of 256 hosts (32% complete)
[*] Scanned  91 of 256 hosts (35% complete)
[*] Scanned 130 of 256 hosts (50% complete)
[*] Scanned 172 of 256 hosts (67% complete)
[*] Scanned 240 of 256 hosts (93% complete)
[*] Scanned 251 of 256 hosts (98% complete)
[*] Scanned 255 of 256 hosts (99% complete)
[*] Scanned 256 of 256 hosts (100% complete)
[*] Auxiliary module execution completed
msf auxiliary(git_scanner) >